sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp

```diff
@@ -134,6 +134,27 @@ namespace Azure { namespace Core { namespace Http {
     bool IgnoreUnknownCertificateAuthority = false;
   };
 
+  namespace _internal {
+    /**
+     * @brief Factory to produce a context with special settings like no client cert context.
+     *
+     * @details Client SDKs can use this class to create an Azure Context to change the
+     * WinHttpTransport default behavior. This is because a Client SDK doesn't know about the
+     * specifics of a transport adapter. So, on Windows, SDK client can set options and only
+     * WinHTTPTransport will check for those options to be in the context.
+     *
+     */
+    class WinHttpTransportContextProvider {
+    public:
+      // Factory with no constructor
+      WinHttpTransportContextProvider() = delete;
+
+      // Creates a set up token to make
+      static Azure::Core::Context GetNoClientCertificateContext(Azure::Core::Context const& parent);
+      static bool HasNoClientCertificateConfiguration(Azure::Core::Context const& context);
```

Comment:
This doesn't need to be public on the class, right? It's looks like it is only a helper used within the transport.

Reply:
Yes, we could make the winHttpTransport adapter to directly call `context.TryGet(..` , but I wanted to delegate all the logic for creating the context and checking for it to one single place.  This would enable a case where some other component (a pipeline policy, a transport adapter, logger, etc.) could also react to a context with this key/value

Reply:
I understand you want to group the logic together in one place. I am trying to reduce the "public" surface area that we can't change or break in the future, and it looks like this method can be made private for now. We can always move it to this class in the future and it be non-breaking.

For example the `NoClientCertificateConfiguration` context key is outside of this class already.

**That said, it's up to you.** The current design with two static methods seems very specific to this use case and option, so not sure how helpful/general purpose this class is to begin with, especially if we want to extend it in the future. Would we just add 2 more methods each time we add a new context key value?